cloudinit/net/sysconfig.py

```diff
@@ -356,9 +358,16 @@ def _render_iface_shared(cls, iface, iface_cfg, flavor):
         flavor_defaults = copy.deepcopy(cls.iface_defaults.get(flavor, {}))
         iface_cfg.update(flavor_defaults)
 
-        for old_key in ('mac_address', 'mtu', 'accept-ra'):
+        for old_key in ('mac_address', 'mtu', 'accept-ra', 'wakeonlan'):
             old_value = iface.get(old_key)
             if old_value is not None:
+                if old_key == 'wakeonlan':
+                    # only set WakeOnLan on physical interfaces
+                    if iface['type'] == 'physical':
+                        new_key = cls.cfg_key_maps[flavor].get(old_key)
+                        if new_key:
+                            iface_cfg[new_key] = "wol g"
+                    continue
```

Comment:
This doesn't feel quite like it fits here, for a couple of reasons. Firstly, the other `old_key`s in this for loop are simple key renames (with one "skip" clause), whereas what we're doing here is setting a static value if a particular key is present: I think there's more code here to fit that different behaviour into the existing structure than we really need. Secondly, and relatedly, the other keys listed in `cfg_key_maps` are 1:1 mappings: the value of "mtu", e.g., should always be the entire value of "MTU" in the rendered output. This isn't true of "wakeonlan" and `ETHTOOL_OPTS`: `ETHTOOL_OPTS` can be used to configure more than just "wakeonlan", so has more of a (potential) many-to-one relationship.

I wonder if doing something like `if "wakeonlan" in iface and iface["wakeonlan"] and iface["type"] == "physical": iface["ETHTOOL_OPTS"] = "wol g"` (outside of the for loop) would do the trick?

(I think this also has the "non-None false values will still enable WOL" issue I identified above; the second clause in this if statement addresses that.)

Reply:
Reworked as recommended.